### PR TITLE
Wp 202 use grunt to build webinos.js

### PR DESCRIPTION
**Pull request by aldafu**
Remove java+nodejs build system to generate webinos.js with grunt.

Grunt is a task based build system based on NodeJS.

Install grunt globally with: npm install -g grunt
This makes sure grunt is added to your PATH.

This commit adds the grunt.js build file which comes with the following tasks:
- default - when called with task name a unminified webinos.js file is built
- minify - builds webinos.js and a minified webinos.min.js.
- clean - removes both generate files from minify task
- clean-certs - removes the certificate directories in ~/.webinos

To generate webinos.js simply run: grunt
in the root directory.

Jira-issue: http://jira.webinos.org/browse/WP-202

### CODE REVIEW

**Comment by habibvirji** [comment]
Resubmit PR as it conflicts with changes submitted by Andre.. ( you need to also include new files added by andre in your grunt file)

Send the contents of the message to the whole WP4 mailing group.

**Comment by aldafu** [comment]
Should have mentioned that I posted this more for review, please don't merge yet. ;)

Andrés files should be included by the way. It's the webnotifications file, no?

**Comment by habibvirji** [comment]
One for more suggestion, as you read .webinos subfolder for clean operation. Would suggest to have same approach to read all subdir of WRT folder, and create file list automatically. It will help in avoiding requirement to add in grunt file manually.

**Comment by habibvirji** [comment]
It is web notification and actuator

**Comment by aldafu** [comment]
Okay, so actuator is missing.

about automatically adding all files from the wrt/ dir. That's not possible as the order in which they are concatenated is very important as they depend in a defined order on each other.

File: grunt.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  var os = require('os');
+  var path = require('path');
+
+  grunt.initConfig({
+    // config props
+    generated: {
+      normal: 'webinos/web_root/webinos.js',
+      min: 'webinos/web_root/webinos.min.js'
+    },
+    exlcudepaths: ['auth_api', 'context_manager', 'logs', 'wrt'],
```
Comment (eelcocramer) [inline grunt.js]: Is exlcude a typo?

File: package.json [hunk 2]
```diff
@@ -40,6 +40,6 @@
   , "webinos_pzp": "webinos_pzp.js"
   }
 , "scripts":
-  { "install": "node-gyp configure build && node webinos_build.js"
+  { "install": "node-gyp configure build"
```
Comment (eelcocramer) [inline package.json]: Shouldn't grunt be called here?

**Comment by aldafu** [comment]
Just pushed some changes, please review and test.

What's new:
- added actuator
- fixed misspelled variable
- call grunt as part of npm install

**Pull request merged**